src/mysql_k8s_helpers.py

```diff
@@ -188,7 +188,13 @@ def fix_data_dir(self, container: Container) -> None:
         Until the ability to set fsGroup and fsGroupChangePolicy via Pod securityContext
         is available we fix permissions incorrectly with chown.
         """
-        paths = container.list_files(MYSQL_DATA_DIR, itself=True)
+        try:
```

Comment:
Instead of try hook if a somewhat generic exception `APIError`, let's test with the [container.exists](https://documentation.ubuntu.com/ops/latest/reference/ops/#ops.Container.exists) method before